examples/cpp/pyperf/PyPerfNativeStackTrace.cc

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) Granulate. All rights reserved.
+ * Licensed under the AGPL3 License. See LICENSE.txt for license information.
+ */
+
+#include "PyPerfNativeStackTrace.h"
+
+#include <sys/uio.h>
+#include <unistd.h>
+
+#include <cstdio>
+#include <cstring>
+
+#include "PyPerfLoggingHelper.h"
+
+namespace ebpf {
+namespace pyperf {
+
+const uint8_t *NativeStackTrace::stack = NULL;
+size_t NativeStackTrace::stack_len = 0;
+uint64_t NativeStackTrace::sp = 0;
+uint64_t NativeStackTrace::ip = 0;
+
+NativeStackTrace::NativeStackTrace(uint32_t pid, const unsigned char *raw_stack,
+                                   size_t stack_len, uint64_t ip, uint64_t sp) : errors(0) {
+  NativeStackTrace::stack = raw_stack;
+  NativeStackTrace::stack_len = stack_len;
+  NativeStackTrace::ip = ip;
+  NativeStackTrace::sp = sp;
+
+  unw_accessors_t my_accessors = _UPT_accessors;
+  my_accessors.access_mem = NativeStackTrace::access_mem;
+  my_accessors.access_reg = NativeStackTrace::access_reg;
+
+  unw_addr_space_t as = unw_create_addr_space(&my_accessors, 0);
+  void *upt = _UPT_create(pid);
+  if (!upt) {
+    logInfo(1, "Failed to _UPT_create\n");
+    this->symbols.push_back(std::string("[Error]"));
```

Comment:
I prefer that we follow what we did in https://github.com/Granulate/bcc/pull/23 - give a better error here, for example `[Error _UPT_create %d]` with the errno (or whatever else libunwind provides, if anything)

Reply:
Same for other error cases, ofc